### ISSUE TITLE
How to deploy toolkit components programatically i.e. using "helm install" ?

### ISSUE THREAD

**Issue by kiich** [fluxcd/helm-controller#211]
I'm following your excellent guide at https://toolkit.fluxcd.io/guides/installation/ and have managed to get up and running with helm-controller and other toolkit components.

This is fantastic and quick on my laptop / dev environment but i couldn't find any documentation on how to roll these out into non-dev clusters such as for prod?

I guess it is indeed a catch 22 because we want to install `helm-controller` that handles helm release using helm - but that's the whole point of helm-controller so i guess that is why the `flux` cli is there for. We typically do not use a cli to release out to prod so was looking to see if there's other ways to do this.

I see `flux install --export` is available which does indeed show me the manifests used?

**Comment by hiddeco** [comment]
Are you only interested in running the `helm-controller` (and `source-controller`) to drive automated Helm releases, or do you want the "GitOps experience" as well (which requires `kustomize-controller`)?

**Comment by kiich** [comment]
Hey @hiddeco 
We are currently looking just to use `helm-controller and `source-controller` for now - so no gitops experience.

After having further look though, we've found https://github.com/fluxcd/helm-controller/tree/main/config which seems to have the manifests files - are these what `flux install --export` produces? Can we just helm package these up (with some additional helm chart files of course)?

**Comment by hiddeco** [comment]
Yes, that is indeed probably what you want to do. To see how the manifests are build, see the following:

https://github.com/fluxcd/helm-controller/blob/main/Makefile#L29-L30

https://github.com/fluxcd/helm-controller/blob/main/Makefile#L37-L39

I would be cautious with deploying the Custom Resource Definitions using Helm however, [as you may know that Helm's lifecycle of those resources is not the best](https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#install-a-crd-declaration-before-using-the-resource), and additions to the definitions are made for MINOR releases.

**Comment by kiich** [comment]
Thanks for your recommendation @hiddeco - we shall refer to the lines.
And thanks for the heads up on CRD via helm - we used to do this via the `crd-install` hook in helm2 but good to know other approaches in helm 3. Both to be fair sounds like another layer to worry about so we might end up going via the `flux` cli in the end (which i assume handles these dependencies correctly?).

But it's good to get the full picture - thanks again. Closing this.
